# 100DayOfSwiftUI - Day 8 - 2022-04-22

Today I learned about default parameters and functions that throw exceptions and how to call them and handle them.


## Code

https://github.com/phptuts/100DaysOfSwiftUI/blob/main/Day8.playground/Contents.swift

## Default Parameters

```swift
func printTimesTable(for number: Int, end: Int = 12) {
    for i in 1...end {
        print("\(i) x \(number) is \(i * number)")
    }
}

```

## Throw Errors

```
enum PasswordError: Error {
    case short, obvious
}

func checkPassword(_ password: String) throws -> String {
    if password.count < 5 {
        throw PasswordError.short
    }
    if password == "12345" {
        throw PasswordError.obvious
        
    }
    
    if password.count < 8 {
        return "OK"
    } else if password.count < 10 {
        return "Good"
    } else {
        return "Excellent"
    }
}
```

## Calling a function that throws an error

```swift
let string = "12345"

do {
    let result = try checkPassword(string)
    print("Password Result: \(result)")
} catch PasswordError.short {
    print("Password too short")
} catch PasswordError.obvious {
    print("Password obvious")
} catch {
    print("There was an error.", error)
}
```
